sonarqube/resource_sonarqube_setting.go

```diff
@@ -306,21 +307,7 @@ func synchronizeSettings(d *schema.ResourceData, m interface{}) (bool, error) {
 }
 
 func checkSettingDiff(a map[string]interface{}, b Setting) bool {
-	if a["value"] != nil {
-		return a["value"].(string) != b.Value
-	} else if a["values"] != nil {
-		// array of strings
-		values := a["field_values"].([]string)
-		if len(values) != len(b.FieldValues) {
-			return false
-		}
-		for i := range values {
-			if string(values[i]) != string(b.Values[i]) {
-				return false
-			}
-		}
-		return true
-	} else if a["field_values"] != nil {
+	if a["field_values"] != nil {
```

Comment:
What are we gaining by refactoring in this way?
I would suggest that it is far far far more likely that a setting is single-valued rather than multi-valued so shouldn't we check that first as the most likely case?  
Aside from it being a micro-optimisation, it makes more sense IMO to handle the most common case as early as possible, which is what the original code was doing

Reply:
What was happening is that terraform loads the value from the setting property from the resource.
But even the field `value` was not being set in the tf script, the value at this point is an empty string instead of being null.

Reply:
so...couldn't you check `if a["value"] != nil && a["value"] != ""` first?
You would possibly also need to account for them all being empty - because settings are cleared out vs "is it value or values or field_values" that has a value - maybe none of them do

Reply:
I though about that, but I think that we cannot guarantee that no one will set the value as "" for a specific setting